src/domain-functions.ts

```diff
@@ -70,10 +71,10 @@ function all<Fns extends DomainFunction[]>(
   ...fns: Fns
 ): DomainFunction<UnpackAll<Fns>> {
   return ((input, environment) => {
-    const [first, ...rest] = fns.map((df) =>
-      A.λ(() => fromSuccess(df)(input, environment)),
+    const composables = fns.map((df) =>
+      A.composable(() => fromSuccess(df)(input, environment)),
     )
-    return dfResultFromcomposable(A.all(first, ...rest))()
+    return dfResultFromcomposable(A.all(...(composables as [Composable])))()
```

Comment:
we need the typecast since `A.all` will not take empty arrays and the map will always resolve to a type that is potentially empty.

Reply:
to make it like a not empty array instead of a tuple of size 1 you can use `[Composable, ...Composable[]]`

Reply:
That was my first intuition as well, but the `AllArguments` bit messes up this case. Perhaps we still need to fix something for cases where I want to pass a list of functions, but since this is not the main use case for compositions (and we are not documenting this yet) I'm happy to merge as is.